pkg/operator/terminationobserver/termination_metrics.go

```diff
@@ -0,0 +1,57 @@
+package terminationobserver
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+type terminationMetrics struct {
+	store           Store
+	apiServerMetric *prometheus.GaugeVec
+}
+
+// RegisterMetrics registers the termination metrics into legacy prometheus registry.
+func RegisterMetrics(store Store) {
```

Comment:
this store does not make it better.

Reply:
it's still a back reference. Why do we need it? Why not call out to the metric from the place where we actually collect the data originally?

Reply:
in other words: make the call graph a tree. No need for any synchronization (locks...).

Reply:
We are not collecting data synchronously. We collect pods from one informer and we collect events for second informer and I use the store to represents that data and make them collectable by prometheus. I don't see any complexity here, there is just simple `map` with synchronized access to read/write (which will be needed regardless of merging the logic of observer and metrics collector).